go/private/rules/test.bzl

```diff
@@ -138,27 +141,25 @@ def _go_test_impl(ctx):
     test_gc_linkopts.extend(["-X", "+initfirst/github.com/bazelbuild/rules_go/go/tools/bzltestutil/chdir.RunDir=" + run_dir])
 
     # Now compile the test binary itself
-    test_library = GoLibrary(
+    test_deps = external_archive.direct + [external_archive] + ctx.attr._testmain_additional_deps
+    if ctx.configuration.coverage_enabled:
+        test_deps.append(go.coverdata)
+    test_go_info = new_go_info(
+        go,
+        struct(
+            deps = test_deps,
+        ),
         name = go.label.name + "~testmain",
-        label = go.label,
         importpath = "testmain",
-        importmap = "testmain",
-        importpath_aliases = (),
         pathtype = INFERRED_PATH,
         is_main = True,
-        resolve = None,
+        generated_srcs = [main_go],
+        coverage_instrumented = False,
```

Comment:
Is this correct here even if coverage is enabled?

Reply:
it seemed off to me as well, but that's what the `False` on line 157 was doing. It might be a preexisting bug, I wasn't sure.

Reply:
Taking a look at this again, which line is this translated from in the original source? If we drop this argument, this line would use the info from the `go_context`:
https://github.com/bazelbuild/rules_go/pull/4030/files#diff-f6aaf6101bc7030a9b8d09d47a1af5918ef80ca113d867cf8b781d8d4d9d5722R318

The only place where `coverage_instrumented = False` was set explicitly appears to be the nogo binary.

Reply:
Clarified this offline: https://github.com/bazelbuild/rules_go/pull/4030/files#diff-2f4977ebad0d8fd012c0d532023ebe9202e795bf502029f7df454d5fc4f96656L165 is the reference and this makes sense as test sources should not be instrumented.